daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -1209,27 +1209,32 @@ toCtors env t = Ctors (getName t) <$> mapM convTypeVar (tyConTyVars t) <*> cs
 --   unpacking them faster in the current intepreter. Since the constructor
 --   names of newtypes are meant to be meaningless, this is acceptable.
 ctorLabels :: TyConFlavour -> DataCon -> [FieldName]
-ctorLabels flv con
-  | flv `elem` [ClassFlavour, TupleFlavour Boxed] || isTupleDataCon con
-    -- NOTE(MH): The line below is a workaround for ghc issue
-    -- https://github.com/ghc/ghc/blob/ae4f1033cfe131fca9416e2993bda081e1f8c152/compiler/types/TyCon.hs#L2030
-    || (is con == "Unit" && moduleNameString (GHC.moduleName (nameModule (getName con))) == "GHC.Tuple")
-  = map (mkField . (:) '_' . show) [1..dataConSourceArity con]
-  | flv == NewtypeFlavour && null lbls
-  = [mkField "unpack"]
-  | otherwise
-  = map convFieldName lbls
-  where lbls = dataConFieldLabels con
+ctorLabels flv con =
+    [mkField $ "_d" <> show i | i <- [1 .. length thetas]] ++ (ctorLabels0 flv con)
+  where
+  thetas = dataConTheta con
+  ctorLabels0 flv con
```

Comment:
I'd have just called this `f` or similar. `ctorLabels0` seems an odd name

Reply:
renamed it.